File: backend/internal/models/favorite.go
package models

import (
	"time"
)

type Favorite struct {
	ID        string    `json:"id"`
	UserID    string    `json:"user_id"`
	SaleID    string    `json:"sale_id"`
	CreatedAt time.Time `json:"created_at"`
}

type FavoriteWithSale struct {
	Favorite
	Sale GarageSale `json:"sale"`
}

File: mobile/lib/models/favorite.dart
class Favorite {
  final String id;
  final String saleId;
  final DateTime createdAt;

  Favorite({
    required this.id,
    required this.saleId,
    required this.createdAt,
  });

  factory Favorite.fromJson(Map<String, dynamic> json) {
    return Favorite(
      id: json['id'] as String,
      saleId: json['sale_id'] as String,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'sale_id': saleId,
      'created_at': createdAt.toIso8601String(),
    };
  }
}

